ext/nokogiri/xml_namespace.c

```diff
@@ -25,7 +25,13 @@
 VALUE cNokogiriXmlNamespace ;
 
 static void
-dealloc_namespace(xmlNsPtr ns)
+_xml_namespace_mark(xmlNsPtr ns)
+{
+  rb_gc_mark((VALUE)ns->_private);
```

Comment:
Since you support compaction, I think it shouldn't be pinned.

```suggestion
  rb_gc_mark_movable((VALUE)ns->_private);
```

Reply:
The problem is actually that we shouldn't be marking here at all! My mistake.